monitor/monitor.go

```diff
@@ -52,6 +50,10 @@ func RunMonitor(ctx context.Context, containerConfig build.ContainerConfig, relo
 		go func() {
 			defer close(doneCh)
 			defer in.Close()
+			go func() {
+				<-ctx.Done()
```

Comment:
This is to handle `appcontext` canceling from `sigint`. Otherwise this goroutine will keep running although everything else shuts down. Worth thinking through this goroutine structure again to make the cancel path more clear and avoid parallel goroutines with no defined shutdown order. Atm there is a chance for a small race if this happens when the initial `Invoke()` call is also in the middle of processing.